src/Skia/Avalonia.Skia/DrawingContextImpl.Effects.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using Avalonia.Media;
+using SkiaSharp;
+
+namespace Avalonia.Skia;
+
+partial class DrawingContextImpl
+{
+    
+    public void PushEffect(IEffect effect)
+    {
+        CheckLease();
+        using var filter = CreateEffect(effect);
+        var paint = SKPaintCache.Shared.Get();
+        paint.ImageFilter = filter;
+        Canvas.SaveLayer(paint);
```

Comment:
Yes, that does make sense, we could implement something like that later. For now, I think, it can be merged as is, since `Effect` will still be a separate property of the Visual and there are no other public APIs being added.

Reply:
Note, that mutable LayerOptions doesn't really make sense for effects, since they are applied to the entire child scene.

Reply:
So LayerOptions would be useful for combining multiple scene properties into one SaveLayer call

Reply:
Y that was my intention 👍